packages/flutter/lib/src/cupertino/radio.dart

```diff
@@ -210,6 +208,13 @@ class CupertinoRadio<T> extends StatefulWidget {
 
   bool get _selected => value == groupValue;
 
+  /// The default mouse cursor of a [CupertinoRadio].
+  static MouseCursor defaultMouseCursor(Function? onChanged) {
```

Comment:
Could this be an actual `WidgetStateProperty<MouseCursor>` instead of a function that returns a MouseCursor?

Reply:
The `mouseCursor` property of Material `Radio` is nullable, so to pass it to `WidgetStateProperty<MouseCursor>?` which is the type of `CupertinoRadio`'s `mouseCursor` it would need to have resolved to a non-null `MouseCursor` which we can then wrap with `WidgetStateProperty.all`. If the `defaultMouseCursor` function returns a `WidgetStateProperty<MouseCursor>`, there would be an extra step to resolve that into a plain non-null `MouseCursor` which would then be wrapped with `WidgetStateProperty.all`.

There might be a better way to do all this though that I am missing.

Reply:
If I'm understanding correctly, try this below and then see my other comment about Material Radio using this.

```dart
  static WidgetStateProperty<MouseCursor> defaultMouseCursor(Function? onChanged) {
    final MouseCursor mouseCursor = (onChanged != null && kIsWeb)
      ? SystemMouseCursors.click
      : SystemMouseCursors.basic;
    return WidgetStateProperty.all<MouseCursor>(mouseCursor);
  }
```